wordpress-mobile/WordPress-iOS
# Issues/7272 split self hosted flow

Refs #7272 
This PR introduces the site address vc -- the first step in the self-hosted login flow.  The seconds step is the username/password form and will be implemented in a subsequent PR.
Requires #7375 to be merged first.

To test:
- Confirm the visual design was correctly implemented.
- Confirm the form is disabled until entering something resembling an URL (very loose rules used)
- Confirm that a valid WordPress url advances to the next form (currently the old self-hosted screen).
- Confirm that an invalid WordPress url prompts to double-check the address and try again.

Needs review: @nheagy 

Screen shots:

![simulator screen shot jun 14 2017 8 45 01 pm](https://user-images.githubusercontent.com/1435271/27161675-a29df3cc-5142-11e7-8006-6b56eabd698e.png)
![simulator screen shot jun 14 2017 8 45 30 pm](https://user-images.githubusercontent.com/1435271/27161677-a4d5a798-5142-11e7-92bf-4bbb87ce813b.png)
![simulator screen shot jun 14 2017 8 45 38 pm](https://user-images.githubusercontent.com/1435271/27161681-a6cc84cc-5142-11e7-8fe6-8ba781f11f07.png)
